prediction_market_agent/tools/endpoints/prediction_prophet/deployment.py

```diff
@@ -0,0 +1,87 @@
+import math
+
+from fastapi import Request, Response
+from langchain_community.callbacks import get_openai_callback
+from modal import App, Secret, web_endpoint
+from prediction_prophet.benchmark.agents import (
+    _make_prediction as prophet_make_prediction,
+)
+from prediction_prophet.functions.research import research as prophet_research
+from pydantic import BaseModel
+
+from prediction_market_agent.tools.endpoints.utils import MODAL_IMAGE
+from prediction_market_agent.utils import APIKeys
+
+MODEL = "gpt-4o"
+keys = APIKeys()
+
+
+class PredictionProphetResponse(BaseModel):
+    p_yes: float
+    cost_usd: float
+
+
+def usd_to_credits(cost_usd: float) -> int:
+    # 1 USD = 100 credits
+    return math.ceil(cost_usd * 100)
+
+
+def predict(question: str) -> Response:
```

Comment:
Ah we have our own endpoint deployments in https://github.com/gnosis/labs-api/, so this would be moved there.